server/routes/shared_api/autocomplete/helpers.py

```diff
@@ -73,6 +73,39 @@ def execute_maps_request(query: str, language: str) -> Dict:
   return json.loads(response.text)
 
 
+def bag_of_letters(text: str) -> Dict:
+  """Creates a bag-of-letters representation of a given string.
+    Returns:
+    dict: A dictionary where keys are letters and values are their counts.
+    """
+  bag = {}
+  for char in text.lower():
+    if char.isalpha():
+      bag[char] = bag.get(char, 0) + 1
+  return bag
+
+
+def off_by_one_letter(str1_word: str, name_word: str) -> bool:
```

Comment:
Have you looked into [Levenshtein distance](https://en.wikipedia.org/wiki/Levenshtein_distance) as another approach? Levenshtein distance takes letter order into account. not sure if that's helpful.

Reply:
Added a TODO to look into it. I ll take another look at it before we talk about launching this :)